client/alloc_runner.go

```diff
@@ -236,13 +242,17 @@ func (r *AllocRunner) syncStatus() error {
 			}
 		}
 	}
+	r.taskStatusLock.RUnlock()
+
+	// Determine the alloc status
+	r.allocLock.Lock()
+	defer r.allocLock.Unlock()
```

Comment:
Soon after this, we are calling the updater method to update the alloc and that makes an RPC call. Have we considered the implications of locking the alloc during the time the rpc happens? On a bad day/network it might be several seconds which means a lot of backed up operations on the client.